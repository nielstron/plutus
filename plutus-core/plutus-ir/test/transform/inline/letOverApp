<<<<<<< HEAD
{- 
Single occurrence let bindings that are functions.
idFun and funApp both get inlined because they only occur once.

let idFun = \y.y
    funApp = \x.idFun
in funApp 5 6
 -}

(let
  (nonrec)
  
  (termbind
    (strict)
    (vardecl idFun (fun (con integer) (con integer)))
    (lam y (con integer) y)
  )
  
  (termbind
    (strict)
    (vardecl funApp (fun (con integer) (fun (con integer) (con integer))))
    (lam x (con integer) idFun)
  )
  
  [ [ funApp (con integer 5) ] (con integer 6) ]
=======
(let
    (nonrec)
    (termbind
        (strict)
        (vardecl idFun (fun (con integer) (con integer)))
        (lam x (con integer) x)
    )
    (termbind
        (strict)
        (vardecl funApp (fun (con integer) (fun (con integer) (con integer))))
        (lam eta (con integer) (lam eta (con integer) [ idFun eta ]))
    )
    [ [ funApp (con integer 5) ] (con integer 6) ]
>>>>>>> 2c8e964d1 (Add over-application test.)
)