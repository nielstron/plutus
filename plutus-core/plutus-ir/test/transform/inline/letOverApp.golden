[
<<<<<<< HEAD
  [ (lam x (con integer) (lam y (con integer) y)) (con integer 5) ]
=======
  [
    (lam
      eta (con integer) (lam eta (con integer) [ (lam x (con integer) x) eta ])
    )
    (con integer 5)
  ]
>>>>>>> 2c8e964d1 (Add over-application test.)
  (con integer 6)
]